src/senaite/app/supermodel/model.py

```diff
@@ -234,8 +235,10 @@ def get(self, name, default=None):
             accessor = field.getAccessor(self.instance)
             accessor_name = accessor.__name__
 
-            # Metadata lookup by accessor name
-            value = getattr(self.brain, accessor_name, _marker)
+            if not api.is_temporary(self.instance):
```

Comment:
Wouldn't be better to use a `_is_temporary` variable here? I would initialize this variable inside the function `_init_with_instance` instead, so `api.is_temporary` would only be called once.

You could even use this `_temporary` variable inside the `brain` property as well, saving you from the call `get_brain_by_uid`

Reply:
I added an internal `is_temporary` method to cache the call to `api.is_temporary` internally.

In the second case we might get a recursion in the case when we have only the brain, because we use in the `instance` property the `brain` property again:

```python
    @property
    def instance(self):
        """Content instance of the wrapped object
        """
        if self._instance is None:
            logger.debug("SuperModel::instance: *Wakup object*")
            self._instance = api.get_object(self.brain)
        return self._instance
```